Core/src/org/sleuthkit/autopsy/datamodel/accounts/Accounts.java

```diff
@@ -1027,74 +1275,85 @@ protected Sheet createSheet() {
-                    }
-                } catch (TskCoreException | SQLException ex) {
-                    LOGGER.log(Level.SEVERE, "Error querying for account artifacts.", ex); //NON-NLS
+        @Subscribe
+        @Override
+        void handleDataAdded(ModuleDataEvent event) {
+            refresh(true);
+        }
+
+        @Override
+        protected boolean createKeys(List<Long> list) {
 
+            String query
+                    = "SELECT blackboard_artifacts.artifact_id " //NON-NLS
+                    + " FROM blackboard_artifacts " //NON-NLS
+                    + "      JOIN blackboard_attributes ON blackboard_artifacts.artifact_id = blackboard_attributes.artifact_id " //NON-NLS
+                    + " WHERE blackboard_artifacts.artifact_type_id = " + BlackboardArtifact.ARTIFACT_TYPE.TSK_ACCOUNT.getTypeID() //NON-NLS
+                    + "     AND blackboard_attributes.attribute_type_id = " + BlackboardAttribute.ATTRIBUTE_TYPE.TSK_CARD_NUMBER.getTypeID() //NON-NLS
+                    + "     AND blackboard_attributes.value_text >= '" + bin.getBINStart() + "' AND  blackboard_attributes.value_text < '" + (bin.getBINEnd() + 1) + "'" //NON-NLS
+                    + getRejectedArtifactFilterClause()
+                    + " ORDER BY blackboard_attributes.value_text"; //NON-NLS
+            try (SleuthkitCase.CaseDbQuery results = skCase.executeQuery(query);
+                    ResultSet rs = results.getResultSet();) {
+                while (rs.next()) {
+                    list.add(rs.getLong("artifact_id")); //NON-NLS
                 }
-                return list;
+            } catch (TskCoreException | SQLException ex) {
+                LOGGER.log(Level.SEVERE, "Error querying for account artifacts.", ex); //NON-NLS
+
             }
+            return true;
+        }
 
-            @Override
-            protected Node[] createNodes(Long artifactID) {
-                if (skCase == null) {
-                    return new Node[0];
-                }
+        @Override
+        protected Node[] createNodesForKey(Long artifactID) {
+            if (skCase == null) {
+                return new Node[0];
+            }
 
-                try {
-                    BlackboardArtifact art = skCase.getBlackboardArtifact(artifactID);
-                    return new Node[]{new AccountArtifactNode(art)};
-                } catch (TskCoreException ex) {
-                    LOGGER.log(Level.WARNING, "Error creating BlackboardArtifactNode for artifact with ID " + artifactID, ex);   //NON-NLS
-                    return new Node[0];
-                }
+            try {
+                BlackboardArtifact art = skCase.getBlackboardArtifact(artifactID);
+                return new Node[]{new AccountArtifactNode(art)};
+            } catch (TskCoreException ex) {
+                LOGGER.log(Level.WARNING, "Error creating BlackboardArtifactNode for artifact with ID " + artifactID, ex);   //NON-NLS
```

Comment:
An exception here would be a database query failure and should be logged as SEVERE.